src/components/structures/RoomStatusBar.tsx

```diff
@@ -142,6 +147,19 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         dis.fire(Action.FocusSendMessageComposer);
     };
 
+    private onRefreshTimelineClick = (): void => {
+        console.log('TODO: Refresh timeline');
+        // TODO: What's the best way to refresh the timeline? Something like
+        // `room.resetLiveTimeline(null, null);` although this just seems to
+        // clear the timeline. I also tried to split out
+        // `scrollbackFromPaginationToken` from the `scrollback` method in to
+        // paginate from the beginning of the room but it's just not right.
```

Comment:
**What's the best way to refresh the timeline?** Something like `room.resetLiveTimeline(null, null);` although this just seems to clear the timeline.

I also tried to split out `scrollbackFromPaginationToken` from the `scrollback` method in to paginate from the beginning of the room and fill in the timeline again after it goes blank but it's just not right.

<details>
<summary>scrollbackFromPaginationToken refactor</summary>

```js
    /**
     * Retrieve older messages from the given room and put them in the timeline.
     *
     * If this is called multiple times whilst a request is ongoing, the <i>same</i>
     * Promise will be returned. If there was a problem requesting scrollback, there
     * will be a small delay before another request can be made (to prevent tight-looping
     * when there is no connection).
     *
     * @param {Room} room The room to get older messages in.
     * @param {Integer} limit Optional. The maximum number of previous events to
     * pull in. Default: 30.
     * @param {module:client.callback} callback Optional.
     * @return {Promise} Resolves: Room. If you are at the beginning
     * of the timeline, <code>Room.oldState.paginationToken</code> will be
     * <code>null</code>.
     * @return {module:http-api.MatrixError} Rejects: with an error response.
     */
    public scrollback(room: Room, limit = 30, callback?: Callback): Promise<Room> {
        if (utils.isFunction(limit)) {
            callback = limit as any as Callback; // legacy
            limit = undefined;
        }
        let timeToWaitMs = 0;

        let info = this.ongoingScrollbacks[room.roomId] || {};
        if (info.promise) {
            return info.promise;
        } else if (info.errorTs) {
            const timeWaitedMs = Date.now() - info.errorTs;
            timeToWaitMs = Math.max(SCROLLBACK_DELAY_MS - timeWaitedMs, 0);
        }

        if (room.oldState.paginationToken === null) {
            return Promise.resolve(room); // already at the start.
        }
        // attempt to grab more events from the store first
        const numAdded = this.store.scrollback(room, limit).length;
        if (numAdded === limit) {
            // store contained everything we needed.
            return Promise.resolve(room);
        }
        // reduce the required number of events appropriately
        limit = limit - numAdded;

        const prom = Promise.resolve().then(async () => {
            try {
                // wait for a time before doing this request
                // (which may be 0 in order not to special case the code paths)
                await sleep(timeToWaitMs)

                await this.scrollbackFromPaginationToken({
                    room,
                    fromToken: room.oldState.paginationToken,
                    direction: Direction.Backward,
                    limit,
                })
                this.ongoingScrollbacks[room.roomId] = null;
                callback?.(null, room);
                return room;
            } catch(err) {
                this.ongoingScrollbacks[room.roomId] = {
                    errorTs: Date.now(),
                };
                callback?.(err);
                throw err;
            }
        });

        info = {
            promise: prom,
            errorTs: null,
        };

        this.ongoingScrollbacks[room.roomId] = info;
        return prom;
    }

    public async scrollbackFromPaginationToken({
        room,
        fromToken,
        direction,
        limit,
    }: {
        room: Room,
        fromToken: string | null,
        direction: Direction,
        limit?: number,
    }) {
        const res: IMessagesResponse = await this.createMessagesRequest(
            room.roomId,
            fromToken,
            limit,
            direction,
        );

        const matrixEvents = res.chunk.map(this.getEventMapper());
        if (res.state) {
            const stateEvents = res.state.map(this.getEventMapper());
            room.currentState.setUnknownStateEvents(stateEvents);
        }

        const [timelineEvents, threadedEvents] = this.partitionThreadedEvents(room, matrixEvents);

        room.addEventsToTimeline(timelineEvents, true, room.getLiveTimeline());
        await this.processThreadEvents(room, threadedEvents, true);

        room.oldState.paginationToken = res.end;
        if (res.chunk.length === 0) {
            room.oldState.paginationToken = null;
        }
        this.store.storeEvents(room, matrixEvents, res.end, true);
    }
```

</details>

Reply:
Assigned review to get some advice for this question ^

Otherwise just need to add tests for this PR